Tests/AppTests/PipelineTests.swift

```diff
@@ -0,0 +1,208 @@
+@testable import App
+
+import SQLKit
+import Vapor
+import XCTest
+
```

Comment:
The tests in this file essentially encode our processing logic as it stands. It's probably easiest if you give these a read through if you agree with the decisions.

Just as a reminder, the stages are:

1. Reconciliation: fetching the master package list and reconciling it against the db (adding or deleting rows in the `packages` table)
1. Ingestion: fetch GH metadata for a set of package candidates returned by `Package.fetchCandidates(app.db, for: .ingestion, limit: 10) ` and create or update rows in `repositories`
1. Analysis: clone/pull repos for a set of package candidates returned by `Package.fetchCandidates(app.db, for: .analysis, limit: 10) ` and create rows in `versions` and `products` according to contents 

The final test, `test_processing_pipeline`, is the most important one: it encodes end-to-end how the whole processing pipeline works over time if the various stages get triggered repeatedly.

(I should probably add this text as a comment to the test itself... :D )

Reply:
Added this to the README in #39. 👍